lib/repl.js

```diff
@@ -282,17 +282,25 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string' &&
+        e.stack.length > 0)) {
```

Comment:
Not sure how robust you want this code to be, but the `typeof e.stack` will crash with

```js
function FakeError() {}
FakeError.prototype = Object.create(Error.prototype, {
  stack: { get() { throw new Error(); } }
});
var e = new FakeError();
```